controller/api/destination/server.go

```diff
@@ -237,6 +245,11 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 /// util ///
 ////////////
 
+type contextToken struct {
+	Ns       string `json:"ns,omitempty"`
+	NodeName string `json:"nodeName,omitempty"`
+}
+
 func nsFromToken(token string) string {
```

Comment:
I think we need to remove the `nsFromToken` function and instead just use the `token.Ns` field.

Reply:
Hm, I have looked at this again. Wouldn't it make sense to keep the function in if the token is not in json format? The (rough) idea I have in mind is:

```go
func parseToken(token string) contextToken {
  if !isValidJSON(token) {
    ns := nsFromToken(token)
    return contextToken{ns}
  }
  unmarshal json
  ...
  return ctxToken
}
```

Reply:
This is purely a stylistic choice at this point, but personally I'd inline the `nsFromToken` logic into `parseToken`.  The reason being that the name `nsFromToken` suggests that that function is the authoritative way to get a namespace from a token.